### PR TITLE
Graphics library david

### PR DESCRIPTION
**Pull request by dalbanhi**
Would appreciate any feedback

### CODE REVIEW

File: src/System.Drawing.Graphics/tests/UnitTests.cs [hunk 1]
```diff
@@ -9,12 +9,189 @@
 public partial class GraphicsUnitTests
 {
     [Fact]
-    public void Test()
```
Comment (axxu) [inline src/System.Drawing.Graphics/tests/UnitTests.cs]: don't get rid of my test plz :'(

File: src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        {
+            get { return _width; }
+        }
+        public int HeightInPixels
+        {
+            get { return _height; }
+        }
+        public PixelFormat PixelFormat
+        {
+            get { return PixelFormat.RGBA; }
```
Comment (whoisj) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: There's likely history here, but why is the format fixed at RGBA? A comment here would likely go a long way.

File: src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        public byte[] Data
+        {
+            get { return _data; }
```
Comment (whoisj) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: Returning the raw data can likely lead to a race condition as image manipulation is often not done on the UI thread while image rendering is. Is that a concern for this class?

File: src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        public int BytesPerPixel
+        {
+            get { return 0; }
```
Comment (whoisj) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: Fixing `BytesPerPixel` at `0` seems... concerning?

File: src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs [hunk 5]
```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+
+namespace System.Drawing.Graphics
+{
+    public enum PixelFormat
```
Comment (whoisj) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: Is there a reason that we're assuming only 32-bit image formats and only these three formats?

File: src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs (hunk 2 above)
Comment (dalbanhi) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: This is just a placeholder for now, until functionality to actually return the appropriate PixelFormat is implemented, will add that comment for sure

Comment (whoisj) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: Have we started enumerating the supported types? If so, there could be issues as some of the types are 96-bit floating color types etc. There's a lot of assumptions in this class (which can be OK) but there are no comments or documentation addressing them.

**Comment by dalbanhi** [comment]
Will add comments to explain placeholders until further functionality is implemented

File: src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            /* Constructors */
+        private Image(int width, int height)
+        {
+            throw new NotImplementedException();
+        }
+        private Image(string filepath)
+        {
+            throw new NotImplementedException();
+        }
+        private Image(object stream)
```
Comment (whoisj) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: Should this be of type `object` or of type `Stream`?

File: src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs (hunk 3 above)
Comment (dsplaisted) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: > Returning the raw data can likely lead to a race condition as image manipulation is often not done on the UI thread while image rendering is. Is that a concern for this class?

I don't think so.  Here we are prototyping a library focused on server-side image manipulation.  See https://github.com/dotnet/corefx/issues/2020 for some of the context.

We do intend for the library to still work in client scenarios, and in that context you'd need to copy an image so you can render one copy and manipulate another.

File: src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+
+namespace System.Drawing.Graphics
+{
+    public enum PixelFormat
+    {
+        ARGB,
```
Comment (KrzysztofCwalina) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: These should follow our casing conventions. See https://msdn.microsoft.com/en-us/library/ms229043(v=vs.110).aspx

File: src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+            /* Factory Methods */
+        public static Image Create(int width, int height)
+        {
+            return new Image(width, height);
+        }
+        public static Image Load(string filePath)
+        {
+            return new Image(filePath);
+        }
+        public static Image Load(object stream)
```
Comment (KrzysztofCwalina) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: Why is it typed as object?

File: src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs [hunk 9]
```diff
@@ -0,0 +1,75 @@
+﻿using System;
```
Comment (KrzysztofCwalina) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: You should add the licensing comment we use in this repo. See the DLLImports file I changed yesterday.

File: src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+        private byte[] _data;
+        private int _width;
+        private int _height;
+
+            /* Properties */
+        public int WidthInPixels
+        {
+            get { return _width; }
+        }
+        public int HeightInPixels
```
Comment (KrzysztofCwalina) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: I wonder if simply Height and Width would not be better.

File: src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs (hunk 7 above)
Comment (dalbanhi) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: Do you mean specifically that ARGB should be Argb? The enum PixelFormat is in the Pascal Casing Convention

Comment (whoisj) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: Argb
Rgba
Cmyk

Comment (KrzysztofCwalina) [inline src/System.Drawing.Graphics/src/System/Drawing/Graphics/Image.cs]: Correct.

**Pull request closed without merging**